### PR TITLE
Added FMA Operator

### PR DESCRIPTION
**Pull request by Benny-Nottonson**
Added the FMA (Fused Multiply Add) operater to the OP struct, added associated tests.

### CODE REVIEW

File: basalt/autograd/ops/basics.mojo [hunk 1]
```diff
@@ -711,3 +711,52 @@ struct RESHAPE:
...
+        @parameter
+        if tensor_id == 0:
+            var res_grad = Tensor[dtype](ug_shape)
+            elwise_op[ug_shape, t2_shape, mul](res_grad, ug, t2)
+            return res_grad ^
+        elif tensor_id == 1:
+            var res_grad = Tensor[dtype](ug_shape)
+            elwise_op[ug_shape, t1_shape, mul](res_grad, ug, t1)
+            return res_grad ^
+        else:
```
Comment (soraros) [inline basalt/autograd/ops/basics.mojo:761]: Should we put a `constrained[False, ...]()` here?

Comment (Benny-Nottonson) [reply basalt/autograd/ops/basics.mojo:761]: Maybe, but I do think we need a wider discussion about how we want to manage Error handling, especially when we get to more constrained operators like Cast

Comment (soraros) [reply basalt/autograd/ops/basics.mojo:761]: For `cast`, I thin constrain can be imposed like `stdlib.SIMD`. `constrained[False]()` is really poor man's unreachable. I'm good with not addressing error handling here.

**Comment by andresnowak** [comment]
I think everything looks good. Just the only thing missing, if you can also add a test in test_ops_torch, that's the only thing missing I would say.

**Review by Benny-Nottonson — COMMENTED** [review]
looks good to me

**Review by andresnowak — APPROVED** [review]

**Pull request merged**